SESHealth/app/src/main/java/team7/seshealthpatient/Activities/SetupActivity.java

```diff
@@ -0,0 +1,159 @@
+package team7.seshealthpatient.Activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.Toast;
+
+import com.google.android.gms.auth.api.Auth;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.common.api.ResultCallback;
+import com.google.android.gms.common.api.Status;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.Calendar;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import team7.seshealthpatient.R;
+
+public class SetupActivity extends AppCompatActivity{
+
+    private FirebaseDatabase database;
+    private DatabaseReference reference;
+    private FirebaseUser user;
+
+    @BindView(R.id.nameET)
+    EditText nameET;
+
+    @BindView(R.id.phoneET)
+    EditText phoneET;
+
+    @BindView(R.id.bDET)
+    EditText bDET;
+
+    @BindView(R.id.allergiesET)
+    EditText allergiesET;
+
+    @BindView(R.id.medicationET)
+    EditText medicationET;
+
+    @BindView(R.id.genderRG)
+    RadioGroup genderRG;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_setup);
+
+        user = FirebaseAuth.getInstance().getCurrentUser();
+
+        database = FirebaseDatabase.getInstance();
+        reference = database.getReference("Users").child(user.getUid());
+
+        ButterKnife.bind(this);
+        textChangedListeners();
+    }
+
+    @OnClick(R.id.setupMain)
+    public void setUserInfo() {
+        if (checkPassed()) {
+            String[] children = {"fullName", "phoneNo", "birthDate",
+                    "allergies", "medication"};
+
+            EditText[] childrenET = {nameET,phoneET,bDET,allergiesET,medicationET};
+
+            for (int i = 0; i < childrenET.length; i++)
+                reference.child(children[i]).setValue(childrenET[i].getText().toString());
+
+            RadioButton radioButton = findViewById(genderRG.getCheckedRadioButtonId());
```

Comment:
You should use BindView for this instead of findviewbyid, it will also auto cast it. Otherwise just remember to explicitly cast it.

Reply:
The findviewbyid for the radiobutton is to set its value to the selected radiobutton. Bindview for that doesn't seem to work, since it states the value needs to be constant.